### PR TITLE
Fixed german translations of registration validation messages

### PR DESCRIPTION
**Pull request by saschaszott**
The translation key `register. messages.validate.login.pattern` was not translated to German.

Furthermore, use `E-Mail` instead of `Email`.

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by saschaszott** [comment]
4 tests are failing with timeout

**Comment by atomfrede** [comment]
Thanks for the PR.

File: generators/languages/templates/src/main/webapp/i18n/de/register.json [hunk 1]
```diff
@@ -10,14 +10,14 @@
...
+                    "pattern": "Ihr Benutzername ist ungültig."
                 }
             },
-            "success": "<strong>Registrierung gespeichert!</strong> Bitte überprüfen Sie Ihre Emails für die Bestätigung.",
+            "success": "<strong>Registrierung gespeichert!</strong> Bitte überprüfen Sie Ihre E-Mails für die Bestätigung.",
             "error": {
                 "fail": "<strong>Registrierung fehlgeschlagen!</strong> Bitte versuchen Sie es später nochmal.",
                 "userexists": "<strong>Benutzername bereits vergeben!</strong> Bitte wählen Sie einen anderen aus.",
-                "emailexists": "<strong>Email wird bereits verwendet!</strong> Bitte wählen Sie eine andere aus."
+                "emailexists": "<strong>E-Mailadresse wird bereits verwendet!</strong> Bitte wählen Sie eine andere aus."
```
Comment (pascalgrimaud) [inline generators/languages/templates/src/main/webapp/i18n/de/register.json:20]: missing space ?

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

File: generators/languages/templates/src/main/webapp/i18n/de/register.json (hunk 1 above)
Comment (atomfrede) [reply generators/languages/templates/src/main/webapp/i18n/de/register.json:20]: You mean `E-Mailadresse`? I think it should be `E-Mail-Adresse` according to the [Duden](https://www.duden.de/rechtschreibung/E_Mail_Adresse)

Comment (pascalgrimaud) [reply generators/languages/templates/src/main/webapp/i18n/de/register.json:20]: yes, you know better than me :-)

Comment (atomfrede) [reply generators/languages/templates/src/main/webapp/i18n/de/register.json:20]: In fact I also had to look it up :)

Comment (saschaszott) [reply generators/languages/templates/src/main/webapp/i18n/de/register.json]: @atomfrede fixed

**Review by pascalgrimaud — APPROVED** [review]

**Comment by atomfrede** [comment]
Thanks again @saschaszott and congrats to your first contribution!

**Review by atomfrede — APPROVED** [review]

**Pull request merged**